server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -834,6 +869,20 @@ private static int computeNumChangedSegments(List<String> segmentIds, int[] segm
     return numChangedSegments;
   }
 
+  private static String getConditionForVersions(
+      final List<String> versions
+  )
+  {
+    if (CollectionUtils.isNullOrEmpty(versions)) {
+      return "";
+    }
+
+    final String versionsCsv = versions.stream()
+                                       .map(version -> "'" + version + "'")
+                                       .collect(Collectors.joining(","));
+    return StringUtils.format(" AND version IN (%s)", versionsCsv);
```

Comment:
Are the versions validated in anyway? Can an attacker construct a versions payload which allows for sort of injection, or what would protect against this?

Reply:
Good call. I have made versions parameterized. Also, I noticed an existing pattern with segment IDs too, which should be changed to the parameterized form as well. I will follow up on that in a separate patch.